libsolidity/interface/CompilerStack.cpp

```diff
@@ -466,7 +467,12 @@ eth::AssemblyItems const* CompilerStack::assemblyItems(string const& _contractNa
 eth::AssemblyItems const* CompilerStack::runtimeAssemblyItems(string const& _contractName) const
 {
 	if (m_stackState != CompilationSuccessful)
-		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("Compilation was not successful."));
+	{
+		if (m_parserErrorRecovery)
```

Comment:
Same here, this has to fail hard.

Reply:
Not sure why I had this. Reviewing things, I don't see why there is a need to. Should be fixed in 2dbe19036

It should (and I believe was) the case that if there are any parsing errors, no bytecode is produced.